corehq/apps/export/models/new.py

```diff
@@ -0,0 +1,370 @@
+        [{'bar': 'a'}, {'bar': 'b'}]
+        >>> TableConfiguration()._get_sub_documents(['foo', 'bar'], [{'foo': [{'bar': {'baz': 'a'}}, {'bar': {'baz': 'b'}},]}]
+        [{'baz': 'a'}, {'baz': 'b'}]
+
+        :param path: A list of a strings
+        :param docs: A list of dicts representing form submissions
+        :return:
+        """
+        if len(path) == 0:
+            return docs
+
+        new_docs = []
+        for doc in docs:
+            next_doc = doc.get(path[0], {})
+            if type(next_doc) == list:
+                new_docs.extend(next_doc)
+            else:
+                new_docs.append(next_doc)
+        return self._get_sub_documents(path[1:], new_docs)
+
+
+class ExportInstance(Document):
+    tables = ListProperty(TableConfiguration)
+
+    class Meta:
+        app_label = 'export'
+
+
+class ExportRow(object):
+    def __init__(self, data):
+        self.data = data
+
+
+class ScalarItem(ExportItem):
+    """
+    A text, numeric, date, etc. question or case property
+    """
+
+
+class Option(DocumentSchema):
+    """
+    This object represents a multiple choice question option.
+
+    last_occurrence is an app build number representing the last version of the app in
+    which this option was present.
+    """
+    last_occurrence = IntegerProperty()
+    value = StringProperty()
+
+
+class MultipleChoiceItem(ExportItem):
+    """
+    A multiple choice question or case property
+    Choices is the union of choices for the question in each of the builds with
+    this question.
+    """
+    options = SchemaListProperty(Option)
+
+    @classmethod
+    def create_from_question(cls, question, appVersion):
+        item = super(MultipleChoiceItem, cls).create_from_question(question, appVersion)
+
+        for option in question['options']:
+            item.options.append(Option(
+                last_occurrence=appVersion,
+                value=option['value']
+            ))
+        return item
+
+    @classmethod
+    def merge(cls, one, two):
+        item = super(MultipleChoiceItem, cls).merge(one, two)
+        options = _merge_lists(one.options, two.options,
+            keyfn=lambda i: i.value,
+            resolvefn=lambda option1, option2:
+                Option(
+                    value=option1.value,
+                    last_occurrence=max(option1.last_occurrence, option2.last_occurrence)
+                ),
+            copyfn=lambda option: Option(option.to_json())
+        )
+
+        item.options = options
+        return item
+
+
+class ExportGroupSchema(DocumentSchema):
+    """
+    An object representing the `ExportItem`s that would appear in a single export table, such as all the
+    questions in a particular repeat group, or all the questions not in any repeat group.
+    """
+    path = ListProperty()
+    items = SchemaListProperty(ExportItem)
+    last_occurrence = IntegerProperty()
+
+
+class ExportDataSchema(DocumentSchema):
```

Comment:
it probably will be, but the models are still in flux a bit. there's a possibility this could be part of the ExportConfiguration